README.rst

```diff
@@ -424,6 +434,105 @@ Keys to take special care are the ones needed to configure Kafka and advertised_
      - ``lowest``
      - Decides on what basis the Karapace cluster master is chosen (only relevant in a multi node setup)
 
+
+Authentication and authorization of Karapace Schema Registry REST API
+=====================================================================
+
+To enable HTTP Basic Authentication and user authorization the authorization configuration file is set in the main configuration key ``registry_authfile`` of the Karapace.
+
+Karapace Schema Registry authorization file is an optional JSON configuration, which contains a list of authorized users in ``users`` and a list of access control rules in ``permissions``.
+
+Each user entry contains following attributes:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``username``
+     - A string
+   * - ``algorithm``
+     - One of supported hashing algorithms, ``scrypt``, ``sha1``, ``sha256``, or ``sha512``
+   * - ``salt``
+     - Salt used for hashing the password
+   * - ``password_hash``
+     - Hash string of the password calculated using given algorithm and salt.
+
+Password hashing can be done using ``karapace_mkpasswd`` tool, if installed, or by invoking directly with ``python -m karapace.auth``. The tool generates JSON entry with these fields. ::
+
+  $ karapace_mkpasswd -u user -a sha512 secret
+  {
+      "username": "user",
+      "algorithm": "sha512",
+      "salt": "iuLouaExTeg9ypqTxqP-dw",
+      "password_hash": "R6ghYSXdLGsq6hkQcg8wT4xkD4QToxBhlp7NerTnyB077M+mD2qiN7ZxXCDb4aE+5lExu5P11UpMPYAcVYxSQA=="
+  }
+
+Supported resource authorization:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Resource
+     - Description
+   * - ``Subject:<subject_name>``
+     - Controls authorization to subject. The ``<subject_name>`` is a regular expression to match against the accessed subject.
+   * - ``Config:``
+     - Controls authorization to global schema registry configuration.
```

Comment:
It'd be more logical to change the order of these two tables. First, explain what the element of each entry are and then explain the format of `resource`.